sdgym/summary.py

```diff
@@ -198,9 +233,45 @@ def make_summary_spreadsheet(file_path):
         'align': 'right'
     })
 
-    add_sheet(summary, 'Single Table (Summary)', writer, cell_fmt, index_fmt, header_fmt)
-    add_sheet(quality, 'Single Table (Quality)', writer, cell_fmt, index_fmt, header_fmt)
-    add_sheet(performance, 'Single Table (Performance)', writer, cell_fmt, index_fmt, header_fmt)
-    add_sheet(error_summary, 'Single Table (Errors Summary)', writer, cell_fmt, index_fmt, header_fmt)
-    add_sheet(error_details, 'Single Table (Errors Detail)', writer, cell_fmt, index_fmt, header_fmt)
+    add_sheet(summary, f'Summary ({modality})', writer, cell_fmt, index_fmt, header_fmt)
+    add_sheet(quality, f'Quality ({modality})', writer, cell_fmt, index_fmt, header_fmt)
+    add_sheet(performance, f'Performance ({modality})', writer, cell_fmt, index_fmt, header_fmt)
+    add_sheet(error_summary, f'Errors Summary ({modality})', writer, cell_fmt, index_fmt,
+              header_fmt)
+    add_sheet(error_details, f'Errors Detail ({modality})', writer, cell_fmt, index_fmt,
+              header_fmt)
+
+
+def make_summary_spreadsheet(results_csv_path, output_path=None, baselines=None,
+                             aws_key=None, aws_secret=None):
+    """Create a spreadsheet document organizing information from results.
+
+    This function creates a ``.xlsx`` file containing information from
+    the results of running ``sdgym.run``. The file contains five sheets
+    for each modality: summary, quality, performance, error summary and
+    error details.
+
+    Args:
+        results_csv_path (str):
+            Path to the csv file containing the results.
+        output_path (str):
+            Path constaining where to store the output spreadsheet.
+            Defaults to {results_csv_path}.xlsx.
+        baselines (dict):
+            Optional dict mapping modalities to a list of baseline
+            model names. If not provided, a default dict is used.
+    """
+    results = read_csv(results_csv_path, aws_key, aws_secret)
+    data = preprocess(results)
+    baselines = baselines or MODALITY_BASELINES
+    output_path = output_path or re.sub('.csv$', '.xlsx', results_csv_path)
+    output = io.BytesIO()
+    writer = pd.ExcelWriter(output) if aws_key and aws_secret else pd.ExcelWriter(output_path)
```

Comment:
`aws_key` and `aws_secret` cannot be used to decide whether we are writing in S3 or not, since an S3 path can be passed without keys to let `boto3` use the system-wide credentials.

To keep things simple, I would just not distinguish between a local or an S3 path here and always write to the `BytesIO` and pass it down to `write_file`, which will decide whether to write to a local or remote file based on the given path.

Reply:
I see, I didn't know that. This is nice though since it will be cleaner in the end